specviz/widgets/workspace.py

```diff
@@ -367,7 +367,18 @@ def compose_filter_string(reader):
         if not file_path:
             return
 
-        self.load_data(file_path, file_loader=loader_name_map[fmt])
+        try:
+            self.load_data(file_path, file_loader=loader_name_map[fmt])
+        except:
+            message_box = QMessageBox()
+            message_box.setText("Error loading data set.")
+            message_box.setIcon(QMessageBox.Critical)
+            message_box.setInformativeText(
+                "{}\n{}".format(
+                    sys.exc_info()[0], sys.exc_info()[1])
+            )
+
+            message_box.exec()
```

Comment:
Perhaps this is too nit picky @drdavella, but shouldn't this go into a different PR?  This is adding a nicer text box warning for failed file loads?

Reply:
This is not adding a message box, it's moving it from the `load_data` method to the `_on_load_data` signal callback.